Jenkinsfile

```diff
@@ -10,25 +34,267 @@ def ircNotification(stage, status) {
+                    throw err
+                }
+            }
+        },
+        push: {
+            node {
+                unstash 'scripts'
+                try {
+                    pushDockerhub('mozorg/bedrock_base')
+                } catch(err) {
+                    ircNotification('Dockerhub Base Push', 'warning')
+                }
+            }
+        }
+    }
+
+    stage ('Unit Test') {
+        parallel build: {
+            node {
+                unstash 'scripts'
+                try {
+                    withEnv(['DOCKER_REPOSITORY=mozorg/bedrock_code']) {
+                        sh 'docker/jenkins/run_tests.sh'
+                    }
+                } catch(err) {
+                    ircNotification('Unit Test Code Image', 'failure')
+                    throw err
+                }
+            }
+        },
+        push: {
+            node {
+                unstash 'scripts'
+                try {
+                    pushDockerhub('mozorg/bedrock_code')
+                } catch(err) {
+                    ircNotification('Dockerhub Code Push', 'warning')
+                }
+            }
+        }
+    }
+
+    stage ('Build L10n') {
+        node {
+            unstash 'scripts'
+            try {
+                buildDockerImage(dockerfile: 'bedrock_l10n', fromDockerfile: 'bedrock_code', script: 'include_l10n.sh')
+            } catch(err) {
+                ircNotification('L10n Build', 'failure')
+                throw err
+            }
+            ircNotification('Docker Builds', 'complete')
+        }
     }
 
-    if ( branch == 'master') {
-        ircNotification('Dev Deploy', 'starting')
-        build 'bedrock_base_image'
+    stage ('Push Images') {
+        // push images to docker hub and internal registries
+        parallel dockerhub: {
+            node {
+                unstash 'scripts'
+                try {
+                    pushDockerhub('mozorg/bedrock_l10n', 'mozorg/bedrock')
+                } catch(err) {
+                    ircNotification('Dockerhub Push Failed', 'warning')
+                }
+            }
+        },
+        registry_usw: {
+            node {
+                unstash 'scripts'
+                try {
+                    pushPrivateReg('usw')
+                } catch(err) {
+                    ircNotification('US-West Registry Push', 'failure')
+                    throw err
+                }
+            }
+        },
+        registry_euw: {
+            node {
+                unstash 'scripts'
+                try {
+                    pushPrivateReg('euw')
+                } catch(err) {
+                    ircNotification('EU-West Registry Push', 'failure')
+                    throw err
+                }
+            }
+        }
+        node {
+            unstash 'scripts'
+            unstash 'tests'
+            // prep for next stage
+            sh 'docker/jenkins/build_integration_test_image.sh'
```

Comment:
This creates a docker image, so as long as it executes on the same machine with the same docker daemon it will be fine. If we move to a distributed executor jenkins setup then we may encounter problems. But I think those problems could be overcome with a shared docker repository for all jenkins nodes.